driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -1059,7 +1072,7 @@ public CCMBridge build() {
       int[] generatedJmxPorts = new int[numNodes];
       for (int i = 0; i < numNodes; i++) {
         if (i >= jmxPorts.length) {
-          generatedJmxPorts[i] = TestUtils.findAvailablePort();
```

Comment:
I can see that `TestUtils.findAvailablePort` is still called in lines 651 and 663. Isn't this a problem?

Reply:
I thought those two should be left as is. 
My reasoning was that `add` calls `execute` which should take care of reserving those two ports right away. The chance of picking two same ports in a row in that code is small enough (1 in ~60k ports - we reach 1% chance for collision at over 600 calls).
To put that in perspective, [picking 10 random ports](https://www.wolframalpha.com/input/?i=BirthdayProblem&assumption=%7B%22F%22%2C+%22BirthdayProblem%22%2C+%22pbds%22%7D+-%3E%2265535%22&assumption=%7B%22F%22%2C+%22BirthdayProblem%22%2C+%22n%22%7D+-%3E%2210%22&assumption=%7B%22C%22%2C+%22BirthdayProblem%22%7D+-%3E+%7B%22Formula%22%7D&assumption=%22FSelect%22+-%3E+%7B%7B%22BirthdayProblem%22%7D%7D) during build() is enough to reach 1 in 1457 chance. With multiple tests and multiple test runs it compounds a little more.

Also `selectedPorts` is not a proper map that maintains all used ports at point in time so I did not want to pollute too much code than necessary with it. I don't really maintain it outside of `build()`.
Should i fix those last two usages anyway?

Reply:
Ok. Makes sense.